### PR TITLE
Nxp 26777 baf idempotency support

### PR DESCRIPTION
**Pull request by pierre-gautier**

### CODE REVIEW

File: nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/computation/FilteringPolicy.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+package org.nuxeo.lib.stream.computation;
+
+/**
+ * @since 11.1
+ */
+public interface FilteringPolicy {
+
+    default boolean shouldSkip(String key, ComputationContext context) {
```
Comment (kevinleturc) [inline nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/computation/FilteringPolicy.java]: We shouldn't have `default` on this method as it's the only contract to implement for this interface.

File: nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/log/internals/InMemoryLogOffsetStorage.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+import org.nuxeo.lib.stream.log.LogOffsetStorage;
+import org.nuxeo.lib.stream.log.LogOffset;
+
+/**
+ * @since 11.1
+ */
+public class InMemoryLogOffsetStorage implements LogOffsetStorage {
+
+    protected Map<String, LogOffset> map = new HashMap<>();
```
Comment (kevinleturc) [inline nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/log/internals/InMemoryLogOffsetStorage.java]: We probably need a `ConcurrentHashMap` here?

File: nuxeo-runtime/nuxeo-stream/src/test/java/org/nuxeo/lib/stream/tests/computation/TestStreamProcessor.java [hunk 3]
```diff
@@ -28,7 +28,6 @@
 import java.time.Duration;
 import java.util.Arrays;
 import java.util.Collections;
-
```
Comment (kevinleturc) [inline nuxeo-runtime/nuxeo-stream/src/test/java/org/nuxeo/lib/stream/tests/computation/TestStreamProcessor.java]: Should be kept.

File: nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/computation/FilteringPolicy.java (hunk 1 above)
Comment (pierre-gautier) [reply nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/computation/FilteringPolicy.java]: Would you prefer a concrete class with default implementation ?

**Pull request closed without merging**